tests/Integration/Behaviour/Features/Context/Domain/CartRuleFeatureContext.php

```diff
@@ -0,0 +1,635 @@
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^I specify that (its) ((active from|active until) "([^"]+)")$/
+     */
+    public function iSpecifyThatItsActiveFromOrUntil(array $properties, DateTime $date, string $property)
+    {
+        $properties[$property] = $date;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^I specify that partial use is (enabled|disabled) for (it)$/
+     */
+    public function iSpecifyThatPartialUseIsDisabledForIt(bool $isPartialUseEnabled, array $properties)
+    {
+        $properties['partial_use'] = $isPartialUseEnabled;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^I specify (its) status as (enabled|disabled)$/
+     */
+    public function iSpecifyItsStatusAsEnabled(array $properties, bool $isEnabled)
+    {
+        $properties['active'] = $isEnabled;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^I specify that (it) (should|should not) be highlighted in cart$/
+     */
+    public function iSpecifyThatItShouldNotBeHighlightedInCart(array $properties, bool $highlightInCart)
+    {
+        $properties['highlight'] = $highlightInCart;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^(its) minimum purchase amount in (currency "[^"]+") is "([^"]+)"$/
+     */
+    public function itsMinimumPurchaseAmountIsInCurrency(array $properties, Currency $currency, $amount)
+    {
+        $properties['minimum_amount'] = (float) $amount;
+        $properties['minimum_amount_currency'] = $currency->id;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^(its) minimum purchase amount is tax (included|excluded)$/
+     */
+    public function itsMinimumPurchaseAmountIsTaxIncluded(array $properties, bool $isTaxIncluded)
+    {
+        $properties['minimum_amount_tax'] = $isTaxIncluded;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^(its) minimum purchase amount is shipping (included|excluded)$/
+     */
+    public function itsMinimumPurchaseAmountIsShippingIncluded(array $properties, bool $isShippingIncluded)
+    {
+        $properties['minimum_amount_shipping'] = $isShippingIncluded;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^(it) gives free shipping$/
+     */
+    public function itGivesFreeShipping(array $properties)
+    {
+        $properties['free_shipping'] = true;
+        SharedStorage::getStorage()->setLatestResource($properties);
+    }
+
+    /**
+     * @Given /^(it) gives a reduction amount of "([^"]+)" in (currency "[^"]+") which is tax (included|excluded) and (applies to (order without shipping|specific product|cheapest product|selected products))$/
```

Comment:
can we do something if PrettyCI will be mad about the length of these lines?
I don't think we can split it into multiple lines...

Reply:
nevermind, PrettyCI is green